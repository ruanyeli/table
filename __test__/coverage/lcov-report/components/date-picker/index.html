<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/date-picker</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> components/date-picker
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/367</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/275</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/55</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/364</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="calendar.js"><a href="calendar.js.html">calendar.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="152" class="abs low">0/152</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="124" class="abs low">0/124</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="152" class="abs low">0/152</td>
	</tr>

<tr>
	<td class="file low" data-value="constant.js"><a href="constant.js.html">constant.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	</tr>

<tr>
	<td class="file low" data-value="date-picker.js"><a href="date-picker.js.html">date-picker.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="42" class="abs low">0/42</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="33" class="abs low">0/33</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="40" class="abs low">0/40</td>
	</tr>

<tr>
	<td class="file low" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	</tr>

<tr>
	<td class="file low" data-value="picker.js"><a href="picker.js.html">picker.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="60" class="abs low">0/60</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="44" class="abs low">0/44</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="60" class="abs low">0/60</td>
	</tr>

<tr>
	<td class="file low" data-value="placements.js"><a href="placements.js.html">placements.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	</tr>

<tr>
	<td class="file low" data-value="range-calendar.js"><a href="range-calendar.js.html">range-calendar.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="20" class="abs low">0/20</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	</tr>

<tr>
	<td class="file low" data-value="range-picker.js"><a href="range-picker.js.html">range-picker.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="65" class="abs low">0/65</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="54" class="abs low">0/54</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="64" class="abs low">0/64</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jan 22 2019 16:22:16 GMT+0800 (CST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
