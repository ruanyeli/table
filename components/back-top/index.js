import * as React from 'react';
import Animate from 'rc-animate';
import addEventListener from 'rc-util/lib/Dom/addEventListener';
import classNames from 'classnames';
import omit from 'omit.js';
import getScroll from '../util/getScroll';
import raf from 'raf';
import s from './style';

const easeInOutCubic = (t, b, c, d) => {
  const cc = c - b;
  t /= d / 2;
  if (t < 1) {
    return cc / 2 * t * t * t + b;
  } else {
    return cc / 2 * ((t -= 2) * t * t + 2) + b;
  }
};

function noop() { }

function getDefaultTarget() {
  return window;
}

// export const BackTopProps = {
//   visibilityHeight,
//   onClick,
//   target?: () => HTMLElement | Window;
//   prefixCls,
//   className?: string;
//   style?: React.CSSProperties;
// }

export default class BackTop extends React.Component {
  static defaultProps = {
    visibilityHeight: 400,
  };

  scrollEvent;

  constructor(props) {
    super(props);
    this.state = {
      visible: false,
    };
  }

  getCurrentScrollTop = () => {
    const getTarget = this.props.target || getDefaultTarget;
    const targetNode = getTarget();
    if (targetNode === window) {
      return window.pageYOffset || document.body.scrollTop || document.documentElement.scrollTop;
    }
    return (targetNode).scrollTop;
  }

  scrollToTop = (e) => {
    const scrollTop = this.getCurrentScrollTop();
    const startTime = Date.now();
    const frameFunc = () => {
      const timestamp = Date.now();
      const time = timestamp - startTime;
      this.setScrollTop(easeInOutCubic(time, scrollTop, 0, 450));
      if (time < 450) {
        raf(frameFunc);
      } else {
        this.setScrollTop(0);
      }
    };
    raf(frameFunc);
    (this.props.onClick || noop)(e);
  }

  setScrollTop(value) {
    const getTarget = this.props.target || getDefaultTarget;
    const targetNode = getTarget();
    if (targetNode === window) {
      document.body.scrollTop = value;
      document.documentElement.scrollTop = value;
    } else {
      (targetNode).scrollTop = value;
    }
  }

  handleScroll = () => {
    const { visibilityHeight, target = getDefaultTarget } = this.props;
    const scrollTop = getScroll(target(), true);
    this.setState({
      visible: scrollTop > visibilityHeight,
    });
  }

  componentDidMount() {
    const getTarget = this.props.target || getDefaultTarget;
    this.scrollEvent = addEventListener(getTarget(), 'scroll', this.handleScroll);
    this.handleScroll();
  }

  componentWillUnmount() {
    if (this.scrollEvent) {
      this.scrollEvent.remove();
    }
  }

  render() {
    const { prefixCls = 'td-back-top', className = '', children } = this.props;
    const classString = classNames(prefixCls, className);

    const defaultElement = (
      <div className={`${prefixCls}-content`}>
        <div className={`${prefixCls}-icon`} />
      </div>
    );

    // fix https://fb.me/react-unknown-prop
    const divProps = omit(this.props, [
      'prefixCls',
      'className',
      'children',
      'visibilityHeight',
      'target',
    ]);

    const backTopBtn = this.state.visible ? (
      <div {...divProps} className={classString} onClick={this.scrollToTop}>
        {children || defaultElement}
      </div>
    ) : null;

    return (
      <Animate component="" transitionName="fade">
        {backTopBtn}
      </Animate>
    );
  }
}
