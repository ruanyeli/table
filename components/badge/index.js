import React from 'react';
import PropTypes from 'prop-types';
import Animate from 'rc-animate';
import ScrollNumber from './ScrollNumber';
import classNames from 'classnames';
import './style';

export default class Badge extends React.Component {
  static defaultProps = {
    prefixCls: 'td-badge',
    scrollNumberPrefixCls: 'td-scroll-number',
    count: null,
    showZero: false,
    dot: false,
    overflowCount: 99,
  };

  static propTypes = {
    count: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
    showZero: PropTypes.bool,
    dot: PropTypes.bool,
    overflowCount: PropTypes.number,
  };

  render() {
    const {
      count,
      showZero,
      prefixCls,
      scrollNumberPrefixCls,
      overflowCount,
      className,
      style,
      children,
      dot,
      status,
      text,
      offset,
      title,
      ...restProps,
    } = this.props;
    let displayCount = count > overflowCount ? `${overflowCount}+` : count;
    const isZero = displayCount === '0' || displayCount === 0;
    const isDot = (dot && !isZero) || status;
    // dot mode don't need count
    if (isDot) {
      displayCount = '';
    }
    const isEmpty = displayCount === null || displayCount === undefined || displayCount === '';
    const hidden = (isEmpty || (isZero && !showZero)) && !isDot;
    const statusCls = classNames({
      [`${prefixCls}-status-dot`]: !!status,
      [`${prefixCls}-status-${status}`]: !!status,
    });
    const scrollNumberCls = classNames({
      [`${prefixCls}-dot`]: isDot,
      [`${prefixCls}-count`]: !isDot,
      [`${prefixCls}-multiple-words`]: !isDot && count && count.toString && count.toString().length > 1,
      [`${prefixCls}-status-${status}`]: !!status,
    });
    const badgeCls = classNames(className, prefixCls, {
      [`${prefixCls}-status`]: !!status,
      [`${prefixCls}-not-a-wrapper`]: !children,
    });
    const styleWithOffset = offset ? {
      marginTop: offset[0],
      marginLeft: offset[1],
      ...style,
    } : style;
    // <Badge status="success" />
    if (!children && status) {
      return (
        <span className={badgeCls} style={styleWithOffset}>
          <span className={statusCls} />
          <span className={`${prefixCls}-status-text`}>{text}</span>
        </span>
      );
    }
    const scrollNumber = hidden ? null : (
      <ScrollNumber
        prefixCls={scrollNumberPrefixCls}
        data-show={!hidden}
        className={scrollNumberCls}
        count={displayCount}
        title={title || count}
        style={styleWithOffset}
      />
    );
    const statusText = (hidden || !text) ? null : (
      <span className={`${prefixCls}-status-text`}>{text}</span>
    );
    return (
      <span {...restProps} className={badgeCls}>
        {children}
        <Animate
          component=""
          showProp="data-show"
          transitionName={children ? `${prefixCls}-zoom` : ''}
          transitionAppear
        >
          {scrollNumber}
        </Animate>
        {statusText}
      </span>
    );
  }
}