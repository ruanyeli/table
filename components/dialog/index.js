/**
 * @Author: xiaofei.wang <wxf>
 * @Date:   2017-06-02 16:32:58
 * @Last modified by:   wxf
 * @Last modified time: 2017-06-02 16:32:58
 */

import Dialog from './dialogWarp';
import Confirm from './confirm';

Dialog.confirm = function(props) {
  const config = Object.assign({
    cancelBtn: true,
    icon: 'exclamation-circle',
    type: 'confirm'
  }, props)
  return Confirm(config)
}

Dialog.success = function(props) {
  const config = Object.assign({
    cancelBtn: false,
    icon: 'check-circle-o',
    type: 'success'
  }, props)
  return Confirm(config)
}

Dialog.error = function(props) {
  const config = Object.assign({
    cancelBtn: false,
    icon: 'close-circle-o',
    type: 'error'
  }, props)
  return Confirm(config)
}

Dialog.warning = function(props) {
  const config = Object.assign({
    cancelBtn: false,
    icon: 'exclamation-circle-o',
    type: 'warning'
  }, props)
  return Confirm(config)
}

export default Dialog;
