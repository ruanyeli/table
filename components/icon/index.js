import React from 'react';
import cn from 'classnames';
import omit from 'lodash/omit';
import './style/index';

export default function Icon(props) {
  const { type, className = '' } = props;
  const iconPrefix = 'y-icon';
  const classString = cn({
    [iconPrefix]: true,
    [`${iconPrefix}-${type}`]: !!type,
  }, className);
  // console.log('classString', classString, props);
  return <i {...omit(props, ['type'])} className={classString} />; // props 中除了type的渲染
}
