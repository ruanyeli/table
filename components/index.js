
export { default as Icon } from './icon';

// export { default as Table } from './table';

// export { default } from './table';
