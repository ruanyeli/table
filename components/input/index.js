import Input from './input';
import Search from './search';
import Group from './group';
import Textarea from './textarea';

Input.Search = Search;
Input.Group = Group;
Input.Textarea = Textarea;

export default Input;
