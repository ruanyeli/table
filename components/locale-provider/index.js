import LocaleProvider from "./LocaleProvider";
import LocaleReceiver from "./LocaleReceiver";

export default LocaleProvider;
export { LocaleReceiver }