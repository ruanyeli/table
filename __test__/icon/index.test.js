import React from 'react';
import { shallow, render, mount } from 'enzyme';
import { Icon } from '../../components';

describe('<Icon />', () => {
  it('should render to a loading icon', () => {
    const wrapper = mount(<Icon type="loading" />);
    expect(wrapper.find('i')).toHaveLength(1);

    expect(wrapper.find('.y-icon')).toHaveLength(1);
    expect(wrapper.find('.y-icon-loading')).toHaveLength(1);
  });

  it('loading icon should match snapshot', () => {
    const wrapper = render(<Icon type="loading" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('style props should be used', () => {
    const wrapper = shallow(<Icon type="clock" style={{ fontSize: '16px', color: 'red' }} />);
    // console.log('style', wrapper.debug(), typeof wrapper.props().style.color);
    expect(wrapper.prop('style').color).toEqual('red');
    expect(wrapper.prop('style')).toEqual({ fontSize: '16px', color: 'red' });
    expect(wrapper).toMatchSnapshot();
  });

  it('classnames props should be used', () => {
    const wrapper = shallow(<Icon type="clock" className="icon" />);
    expect(wrapper.hasClass('icon')).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
