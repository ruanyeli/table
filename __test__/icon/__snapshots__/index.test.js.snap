// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> classnames props should be used 1`] = `
<i
  className="y-icon y-icon-clock icon"
/>
`;

exports[`<Icon /> loading icon should match snapshot 1`] = `
<i
  class="y-icon y-icon-loading"
/>
`;

exports[`<Icon /> style props should be used 1`] = `
<i
  className="y-icon y-icon-clock"
  style={
    Object {
      "color": "red",
      "fontSize": "16px",
    }
  }
/>
`;
